File: CS/AuthenticationOwin.Module.Web/Security/CustomSecurityStrategyComplex.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Web;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Security;

namespace AuthenticationOwin.Module.Web.Security
{
    public class CustomSecurityStrategyComplex : SecurityStrategyComplex
    {
        protected override void InitializeNewUserCore(IObjectSpace objectSpace, object user)
        {
            base.InitializeNewUserCore(objectSpace, user);
        }
        public void InitializeNewUser(IObjectSpace objectSpace, object user)
        {
            InitializeNewUserCore(objectSpace, user);
        }
        public override void Logoff()
        {
            if (HttpContext.Current.Request.Cookies[".AspNet.External"] != null)
            {
                HttpContext.Current.Response.Cookies[".AspNet.External"].Expires = DateTime.Now.AddDays(-1);
            }
            base.Logoff();
        }
    }
}

File: VB/AuthenticationOwin.Module.Web/Security/CustomSecurityStrategyComplex.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Web
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Security

Namespace AuthenticationOwin.Module.Web.Security
	Public Class CustomSecurityStrategyComplex
		Inherits SecurityStrategyComplex
		Protected Overrides Sub InitializeNewUserCore(ByVal objectSpace As IObjectSpace, ByVal user As Object)
			MyBase.InitializeNewUserCore(objectSpace, user)
		End Sub
		Public Sub InitializeNewUser(ByVal objectSpace As IObjectSpace, ByVal user As Object)
			InitializeNewUserCore(objectSpace, user)
		End Sub
		Public Overrides Sub Logoff()
			If HttpContext.Current.Request.Cookies(".AspNet.External") IsNot Nothing Then
				HttpContext.Current.Response.Cookies(".AspNet.External").Expires = DateTime.Now.AddDays(-1)
			End If
			MyBase.Logoff()
		End Sub
	End Class
End Namespace
